weblab/datasets/views.py

```diff
@@ -486,3 +487,10 @@ def post(self, request, *args, **kwargs):
             return HttpResponseRedirect(self.get_success_url())
         else:
             return self.render_to_response(self.get_context_data(forms=forms))
+
+
+class DatasetCollaboratorsView(EntityCollaboratorsView):
```

Comment:
Ah, I've realised an issue with just inheriting here. `EntityCollaboratorsView` inherits from `EntityTypeMixin` which we *don't* want to do. So possibly we need another mixin created in `entities/views.py` that contains the common fields/methods, and which both concrete views can inherit from.

So perhaps a new `EditCollaboratorsAbstractView` class, which contains everything from `EntityCollaboratorsView` except:
* no `template_name` field
* get_context_data is changed to remove the `kwargs['type'] = self.object.entity_type` line

Then we could have
```python
class EntityCollaboratorsView(EntityTypeMixin, EditCollaboratorsAbstractView):
    template_name = 'entities/entity_collaborators_form.html'

    def get_context_data(self, **kwargs):
        kwargs['type'] = self.object.entity_type
        return super().get_context_data(**kwargs)
```
and
```python
class DatasetCollaboratorsView(EditCollaboratorsAbstractView):
    template_name = 'datasets/dataset_collaborators_form.html'
```

Reply:
thanks. Ive implemented that but I now get the following error:

`DatasetCollaboratorsView is missing a QuerySet. Define DatasetCollaboratorsView.model, DatasetCollaboratorsView.queryset, or override DatasetCollaboratorsView.get_queryset().`

Reply:
Ah yes, that's given by EntityTypeMixin for the other view; we need to set `model = Dataset` as a field for the datasets one.